azure/scope/cluster.go

```diff
@@ -379,12 +379,15 @@ func (s *ClusterScope) SubnetSpecs() []azure.ResourceSpecGetter {
 			Role:              subnet.Role,
 			NatGatewayName:    subnet.NatGateway.Name,
 		}
+		if s.IsVnetManaged() {
```

Comment:
Do we want to check if the length of `subnet.ServiceEndpoints` is greater than zero before entering into this func and performing `make([]infrav1.ServiceEndpointSpec, 0, len(subnetServiceEndpoints))`? Doing it this way essentially assigns the zero value to `subnetSpec.ServiceEndpoints` when there are no `ServiceEndpoints` defined, but at an extra cost.

Reply:
It needs to be able to be set to an empty slice in the case of someone switching from having service endpoints defined to not having any service endpoints defined. If I add the conditional it won't remove the service endpoints in this scenario.

Reply:
Thanks for thinking of all cluster configuration mutation scenarios!